File: java/resistor-color-duo/src/main/java/ResistorColorDuo.java
import java.util.Arrays;

class ResistorColorDuo {

     final String[] Colors = {"black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"};
    
    int value(String[] colors) {
        if(colors.length > 1)
        return  Arrays.asList(Colors).indexOf(colors[0])*10 + Arrays.asList(Colors).indexOf(colors[1]);
        else 
        throw new IllegalArgumentException("Put two colors or more.");
    }
}

File: typescript/resistor-color-duo/resistor-color-duo.ts
export function decodedValue(colorArray: Array<string>) {
    const colorAry = [
        'black',
        'brown',
        'red',
        'orange',
        'yellow',
        'green',
        'blue',
        'violet',
        'grey',
        'white',
    ];
  const firstNumber: number = colorAry.indexOf(colorArray[0], 0);
  const secondNumber: number = colorAry.indexOf(colorArray[1], 0);
  return (firstNumber*10+secondNumber);
};